sovrin_node/persistence/idr_cache.py

```diff
@@ -91,18 +94,28 @@ def close(self):
         self._keyValueStorage.close()
 
     def currentBatchCreated(self, stateRoot):
```

Comment:
Can we write a more low-level test that ensures that a sequence of 
currentBatchCreated, batchRejected and onBatchCommitted calls work?
Example:
Test1: - sunny day
- save current state and current unCommitted 
- call currentBatchCreated twice (with different states)
- call onBatchCommitted for the first state
- make sure the state is committed and the unCommitted state is equal to the second state
- call onBatchCommitted for the second state
- make sure the state is committed and unCommitted is empty

Test2: - commit not the first state
- save current state and current unCommitted 
- call currentBatchCreated twice (with different states)
- call onBatchCommitted for the second state
- make sure assertion error is raised

Test3: - reject state
- save current state and current unCommitted 
- call currentBatchCreated twice (with different states)
- call batchRejected 
- make sure the second state is rejected

test4:
-